frontend/src/queries/nodes/InsightQuery/utils/filtersToQueryNode.ts

```diff
@@ -96,6 +105,74 @@ export const cleanHiddenLegendSeries = (
         : undefined
 }
 
+const cleanProperties = (parentProperties: FilterType['properties']): InsightsQueryBase['properties'] => {
+    if (!parentProperties || !parentProperties.values) {
+        return parentProperties
+    }
+
+    const processAnyPropertyFilter = (filter: AnyPropertyFilter): AnyPropertyFilter => {
+        if (
+            filter.type === PropertyFilterType.Event ||
+            filter.type === PropertyFilterType.Person ||
+            filter.type === PropertyFilterType.Element ||
+            filter.type === PropertyFilterType.Session ||
+            filter.type === PropertyFilterType.Group ||
+            filter.type === PropertyFilterType.Feature ||
+            filter.type === PropertyFilterType.Recording
```

Comment:
It might make sense to keep the typescript and python conversion as similar as possible to each other, so that we can easily understand differences. In python we have `is_property_with_operator` helper for this: https://github.com/PostHog/posthog/blob/c95b9bced4e295e9d32238437fcd3e75f51a1c4f/posthog/hogql_queries/legacy_compatibility/filter_to_query.py#L52-L53